sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -0,0 +1,113 @@
+           (1e-2, 0.1, 1e2) * RBF((1e-3, 0.1, 1e3)) + (None, 0.0, None)]
+
+def test_gpr_interpolation():
+    """Test the interpolating property for different kernels."""
+    for kernel in kernels:
+        kernel = deepcopy(kernel)
+        gpr = GaussianProcessRegression(kernel=kernel).fit(X, y)
+        y_pred, y_cov = gpr.predict(X, return_cov=True)
+
+        assert_true(np.allclose(y_pred, y))
+        assert_true(np.allclose(np.diag(y_cov), 0.))
+
+
+def test_lml_improving():
+    """ Test that hyperparameter-tuning improves log-marginal likelihood. """
+    for kernel in kernels:
+        if not kernel.has_bounds:
+            continue
+        kernel = deepcopy(kernel)
+        params_initial = kernel.params
+        gpr = GaussianProcessRegression(kernel=kernel).fit(X, y)
+        assert_greater(gpr.log_marginal_likelihood(kernel.params),
+                       gpr.log_marginal_likelihood(params_initial))
+
+
+def test_converged_to_local_maximum():
+    """ Test that we are in local maximum after hyperparameter-optimization. """
+    for kernel in kernels:
+        if not kernel.has_bounds:
+            continue
+        kernel = deepcopy(kernel)
+        gpr = GaussianProcessRegression(kernel=kernel).fit(X, y)
+
+        lml, lml_gradient = gpr.log_marginal_likelihood(kernel.params, True)
+
+        assert_almost_equal(lml_gradient, 0, 5)
+
+
+def test_lml_gradient():
+    """ Compare analytic and numeric gradient of log marginal likelihood. """
+    for kernel in kernels:
+        kernel = deepcopy(kernel)
+        params = kernel.params
+        gpr = GaussianProcessRegression(kernel=kernel).fit(X, y)
+
+        lml, lml_gradient = gpr.log_marginal_likelihood(params, True)
+        lml_gradient_approx = \
+            approx_fprime(params,
+                          lambda theta: gpr.log_marginal_likelihood(theta,
+                                                                    False),
+                          1e-10)
+
+        assert_almost_equal(lml_gradient, lml_gradient_approx, 3)
+
+
+def test_prior():
+    """ Test that GP prior has mean 0 and identical variances."""
+    for kernel in kernels:
+        kernel = deepcopy(kernel)
+        gpr = GaussianProcessRegression(kernel=kernel)
+
+        y_mean, y_cov = gpr.predict(X, return_cov=True)
+
+        assert_almost_equal(y_mean, 0, 5)
+        if len(kernel.params) > 1:
+            # XXX: quite hacky, works only for current kernels
+            assert_almost_equal(np.diag(y_cov), kernel.params[0] , 5)
+        else:
+            assert_almost_equal(np.diag(y_cov), 1, 5)
+
+
+def test_sample_statistics():
+    """ Test that statistics of samples drawn from GP are correct."""
+    for kernel in kernels:
+        kernel = deepcopy(kernel)
+        gpr = GaussianProcessRegression(kernel=kernel).fit(
+            X, y)
+
+        y_mean, y_cov = gpr.predict(X2, return_cov=True)
+
+        samples = gpr.sample(X2, 1000000)
```

Comment:
Unfortunately not. The number of digits in assert_almost_equal could be further reduced if the runtime of this test becomes prohibitive